compiler/src/iree/compiler/Codegen/Common/test/canonicalize_interface_load_store.mlir

```diff
@@ -58,13 +58,13 @@ func.func @dont_fold_reshape_with_not_full_load() {
   %c3 = arith.constant 3 : index
   %c96 = arith.constant 96 : index
   %1 = hal.interface.binding.subspan layout(#pipeline_layout) binding(0) : !flow.dispatch.tensor<readonly:tensor<6x3x1x96xf32>>
-  %2 = hal.interface.binding.subspan layout(#pipeline_layout) binding(0) : !flow.dispatch.tensor<writeonly:tensor<3x3x96xf32>>
-  %3 = flow.dispatch.tensor.load %1, offsets = [%c3, %c0, %c0, %c0], sizes = [%c3, %c3, %c1, %c96], strides = [%c1, %c1, %c1, %c1] : !flow.dispatch.tensor<readonly:tensor<6x3x1x96xf32>> -> tensor<3x3x1x96xf32>
+  %2 = hal.interface.binding.subspan layout(#pipeline_layout) binding(0) : !flow.dispatch.tensor<writeonly:tensor<4x3x96xf32>>
+  %3 = flow.dispatch.tensor.load %1, offsets = [3, 0, 0, 0], sizes = [3, 3, 1, 96], strides = [1, 1, 1, 1] : !flow.dispatch.tensor<readonly:tensor<6x3x1x96xf32>> -> tensor<3x3x1x96xf32>
```

Comment:
This was the bug. when sizes/offsets were index attributes, the pattern did not look at the values. This test was passing because it was using arith.constants instead of index attrs, and the pattern failed any time there were non index attr sizes/offsets.